frb_codegen/src/generator/rust/ty_opaque.rs

```diff
@@ -0,0 +1,170 @@
+        Acc {
+            io: Some(
+                "unsafe {
+                support::opaque_from_dart(self.ptr as _)
+            }"
+                .into(),
+            ),
+            ..Default::default()
+        }
+    }
+
+    /// Handles JsValue to Self conversions.
+    fn wire2api_jsvalue(&self) -> Option<Cow<str>> {
+        #[cfg(target_arch = "wasm64")]
+        {
+            panic!("The wasm64 arch is not supported.");
+        }
+
+        Some(
+            "unsafe {
+                support::opaque_from_dart((self.as_f64().unwrap() as usize) as _)
+            }"
+            .into(),
+        )
+    }
+
+    fn wire_struct_fields(&self) -> Option<Vec<String>> {
+        Some(vec!["ptr: *const core::ffi::c_void".to_owned()])
+    }
+
+    fn static_checks(&self) -> Option<String> {
+        None
+    }
+
+    fn wrapper_struct(&self) -> Option<String> {
+        None
+    }
+
+    fn self_access(&self, obj: String) -> String {
+        obj
+    }
+
+    fn wrap_obj(&self, obj: String) -> String {
+        obj
+    }
+
+    fn convert_to_dart(&self, obj: String) -> String {
+        format!("{}.into_dart()", obj)
+    }
+
+    fn structs(&self) -> String {
+        "".to_owned()
+    }
+
+    fn allocate_funcs(
+        &self,
+        collector: &mut ExternFuncCollector,
+        _block_index: BlockIndex,
+    ) -> Acc<Option<String>> {
+        let rust_wire = self.ir.rust_wire_type(crate::target::Target::Io);
+
+        Acc {
+            io: Some(collector.generate(
+                &format!("new_{}", self.ir.safe_ident()),
+                NO_PARAMS,
+                Some(&format!(
+                    "{}{}",
+                    self.ir.rust_wire_modifier(crate::target::Target::Io),
+                    rust_wire
+                )),
+                &format!("{}::new_with_null_ptr()", rust_wire,),
+                crate::target::Target::Io,
+            )),
+            ..Default::default()
+        }
+    }
+
+    fn opaque_drop_funcs(
+        &self,
+        collector: &mut ExternFuncCollector,
+        _block_index: BlockIndex,
+    ) -> Acc<Option<String>> {
+        Acc {
+            io: Some(collector.generate(
+                &format!("drop_opaque_{}", self.ir.safe_ident()),
+                [("ptr: *const c_void", "")],
+                None,
+                &format!(
+                    "unsafe {{Arc::<{}>::decrement_strong_count(ptr as _);}}",
+                    self.ir.inner_rust
+                ),
+                crate::target::Target::Io,
+            )),
+            wasm: Some(collector.generate(
+                &format!("drop_opaque_{}", self.ir.safe_ident()),
+                [("ptr: *const c_void", "")],
+                None,
+                &format!(
+                    "unsafe {{Arc::<{}>::decrement_strong_count(ptr as _);}}",
+                    self.ir.inner_rust
+                ),
+                crate::target::Target::Wasm,
+            )),
```

Comment:
looks quite similar (or even identical?), can we extract common parts?

Reply:
we can do it:
```
        let mut generate_impl = |target| {
            collector.generate(
                &format!("drop_opaque_{}", self.ir.safe_ident()),
                [("ptr: *const c_void", "")],
                None,
                &format!(
                    "unsafe {{Arc::<{}>::decrement_strong_count(ptr as _);}}",
                    self.ir.inner_rust
                ),
                target,
            )
        };
        Acc {
            io: Some(generate_impl(crate::target::Target::Io)),
            wasm: Some(generate_impl(crate::target::Target::Wasm)),
            ..Default::default()
        }
```